xfields/beam_elements/beambeam_src/beambeam3d.h

```diff
@@ -6,23 +6,34 @@
 #ifndef XFIELDS_BEAMBEAM3D_H
 #define XFIELDS_BEAMBEAM3D_H
 
+//#if !defined(dump_path)
+//    #define dump_path "/Users/pkicsiny/phd/cern/xsuite/outputs/n74c" 
+//#endif
 
 /*gpufun*/
 void synchrobeam_kick(
-        BeamBeamBiGaussian3DData el, const int i_slice,
+        BeamBeamBiGaussian3DData el, LocalParticle *part, 
+        const int64_t flag_beamstrahlung, const int64_t flag_luminosity,
+        BeamBeamBiGaussian3DRecordData beamstrahlung_record, RecordIndex beamstrahlung_table_index, BeamstrahlungTableData beamstrahlung_table,
+        BeamBeamBiGaussian3DRecordData luminosity_record, RecordIndex luminosity_table_index, LuminosityTableData luminosity_table,
+        const int i_slice,
         double const q0, double const p0c,
         double* x_star,
         double* px_star,
         double* y_star,
         double* py_star,
         double* zeta_star,
-        double* pzeta_star){
+        double* pzeta_star, double const tan_phi){
 
     // Get data from memory
     const double q0_bb  = BeamBeamBiGaussian3DData_get_other_beam_q0(el);
     const double min_sigma_diff = BeamBeamBiGaussian3DData_get_min_sigma_diff(el);
     const double threshold_singular = BeamBeamBiGaussian3DData_get_threshold_singular(el);
 
+    // debugging
+    const int64_t turn    = BeamBeamBiGaussian3DData_get_turn(el);
+//    const double flat_y_i = BeamBeamBiGaussian3DData_get_flat_y_i(el); 
+
```

Comment:
Remember to remove this

Reply:
removed turn, flat_y, lumi table and tan_phi.